src/JsonAssetEnqueuer.php

```diff
@@ -4,91 +4,144 @@
+        if (!empty($src)) {
+            $this->wordpressAssetGateway->registerScript('vendor', $src, $dependencies, null, true);
+        }
+
+        foreach ($this->manifest->js as $group => $scripts) {
+            $dependencies = $this->computeDependencyArray($group);
 
-        foreach ($groupNames as $group) {
-            $dependencies = $this->computeDependencyArray($group, $availableGroups);
-            $src          = $this->blogUrl . str_replace($this->container['wwp.asset.folder.prefix'], '', $this->manifest->site->assets_dest) . '/js/' . $group . $versionNum . '.js';
-            wp_enqueue_script($group, $src, $dependencies, null, true);
+            $src = $this->getUrlSrcFrom('js', $group);
+
+            if (!empty($src)) {
+                $this->wordpressAssetGateway->registerScript($group, $src, $dependencies, null, true);
+            }
         }
     }
 
-    protected function computeDependencyArray($groupName, $availableGroups)
+    /**
+     * @param string $groupName
+     * @return string[]
+     */
+    protected function computeDependencyArray(string $groupName)
     {
         $dependencyArray = isset($this->manifest->jsDependencies->{$groupName}) ? $this->manifest->jsDependencies->{$groupName} : [];
 
         return $dependencyArray;
     }
 
     /** @inheritdoc */
-    public function enqueueCritical(array $groupNames)
-    {
-        $versionNum = $this->getVersion();
-        foreach ($groupNames as $group) {
-            if (property_exists($this->manifest->js, $group)) {
-                $src = $_SERVER['DOCUMENT_ROOT'] . str_replace($this->container['wwp.asset.folder.prefix'], '', $this->manifest->site->assets_dest) . '/js/' . $group . $versionNum . '.js';
-                if (file_exists($src)) {
-                    $content = apply_filters('wwp.enqueur.critical.js.content', file_get_contents($src));
-                    if (!empty($content)) {
-                        echo '<script id="critical-js">
-                                ' . $content . '
-                            </script>';
-                    }
-                }
+    public function enqueueStyleGroup(string $groupName)
+    {
+        $this->enqueueStyle($groupName);
+    }
+
+    /** @inheritdoc */
+    public function enqueueScriptGroup(string $groupName)
+    {
+        $this->enqueueScript($groupName);
+    }
+
+    /** @inheritDoc */
+    public function enqueueStyle(string $handle)
+    {
+        $this->wordpressAssetGateway->enqueueStyle($handle);
+    }
+
+    /** @inheritDoc */
+    public function enqueueScript(string $handle)
+    {
+        $this->wordpressAssetGateway->enqueueScript($handle);
+    }
+
+    /** @inheritDoc */
+    public function inlineStyle(string $handle)
+    {
+        if ($this->isPropertyExistInManifest('css', $handle)) {
+            $src = $this->getPathSrcFrom('css', $handle);
+
+            if (file_exists($src) && !$this->wordpressAssetGateway->isAdmin()) {
```

Comment:
C'est forcément une opinion wdf ça, à enlever du niveau framework.
```suggestion
            if (file_exists($src)) {
```